pkg/daemon/server/service/pipeline_watermark_query.go

```diff
@@ -118,7 +118,10 @@ func (ps *pipelineMetadataQuery) GetVertexWatermark(ctx context.Context, request
 		log.Errorf("watermark fetcher not available for vertex %s in the fetcher map", vertexName)
 		return nil, fmt.Errorf("watermark not available for given vertex, %s", vertexName)
 	}
-	vertexWatermark := time.Time(vertexFetcher.GetHeadWatermark()).Unix()
+
+	vertexWatermark := vertexFetcher.GetHeadWatermark().Unix()
+	log.Infof("Head watermark retrieved for vertex %s is %d.", vertexName, vertexWatermark)
```

Comment:
Thanks! I was using level INFO during local development because otherwise I couldn't see the logs. Do you know how to tell my local env to show DEGUB level logs? (set EnvVar NUMAFLOW_DEBUG?)

I set it to INFO level as I think it could be useful informations to show in PROD, which help us debugging at PROD stage. The tradeoff is that we might flood the PROD memory with a lot of logs. Do we have a standard within the team on when to use what log level?

Reply:
We do not log repeated events over and over as Info as it spoils the log file. Makes debugging even tougher. Usually, `Info` is once in a while, mostly while bootup and shutdown. `Error` can be logged often, but in some cases, we group and summarize if the errors that occur are the same (e.g., buffer-full errors).

while debugging you can set `os.SetEnv("NUMAFLOW_DEBUG", "true")` in your test function.

Reply:
Cool got it! Thanks!